### ISSUE TITLE
dotlockfile(1) swallows EPERM errors on lockfile removal

### ISSUE THREAD

**Issue by strugee** [miquels/liblockfile#4]
Testcase:

    $ ./dotlockfile foo.lock sudo chattr +i foo.lock
    $ echo $?
    0
    $ file foo.lock
    foo.lock: ASCII text
    $ rm foo.lock
    rm: cannot remove 'foo.lock': Operation not permitted

**Comment by miquels** [comment]
Yes, but what else could it do? The lock succeeded, the command was run. The lockfile was left laying around, but that is a problem for the _next_ time it runs. But, I do agree that if it then detects that the lockfile is stale but fails to remove it, it should print an error and exit with an appropriate status.

**Comment by miquels** [comment]
I've added https://github.com/miquels/liblockfile/commit/60cf79ad67b945f8f219f2feebb9b20a707f76bc so that we can at least fail fast, and detect internally that this was not just "can't lock". `dotlockfile` should really output warnings/errors if it cannot create a lock because of these sort of errors. However, I hope that I can release a new version that will make it into debian before the freeze, and that is too big of a change - will have to be done later this year.
